src/System.Console/src/System/IO/SyncTextReader.cs

```diff
@@ -31,8 +31,8 @@ protected override void Dispose(bool disposing)
             {
                 lock (_methodLock)
                 {
-                    // Explicitly pick up a potentially methodimpl'ed Dispose
-                    ((IDisposable)_in).Dispose();
+                    // This will always pick up TextReader.Dispose
```

Comment:
I wonder why the author of the original code was worried about the case where a subclass had re-implemented IDisposable. I much prefer what you have, but this is technically an externally observable in that pathological case and there's a cryptic comment showing intention for the old behavior. cc @davkean 

I wouldn't bother commenting the plain Dispose at all.